Documentation/concepts/networking/routing.rst

```diff
@@ -0,0 +1,309 @@
+  simplifies communication of pod traffic within VPCs and avoids the need for
+  SNAT.
+
+* Pod IPs are assigned a security group. The security groups for pods are
+  configured per node which allows to create node pools and give different
+  security group assignments to different pods. See section :ref:`ipam_eni` for
+  more details.
+
+Disadvantages of this model
+---------------------------
+
+* The number of ENI IPs is limited per instance. The limit depends on the EC2
+  instance type. This can become a problem when attempting to run a larger
+  number of pods on very small instance types.
+
+* Allocation of ENIs and ENI IPs requires interaction with the EC2 API which is
+  subject to rate limiting. This is primarily mitigated via the operator
+  design, see section :ref:`ipam_eni` for more details.
+
+Architecture
+------------
+
+Ingress
+~~~~~~~
+
+1. Traffic is received on one of the ENIs attached to the instance which is
+   represented on the node as interface ``ethN``.
+
+2. An IP routing rule ensures that traffic to all local pod IPs is done using
+   the main routing table:
+
+   .. code-block:: bash
+
+       20:	from all to 192.168.105.44 lookup main
+
+3. The main routing table contains an exact match route to steer traffic into a
+   veth pair which is hooked into the pod:
+
+   .. code-block:: bash
+
+       192.168.105.44 dev lxc5a4def8d96c5
+
+4. All traffic passing ``lxc5a4def8d96c5`` on the way into the pod is subject
+   to Cilium's BPF program to enforce network policies, provide service reverse
+   load-balancing, and visibility.
+
+Egress
+~~~~~~
+
+1. The pod's network namespace contains a default route which points to the
+   node's router IP via the veth pair which is named ``eth0`` inside of the pod
+   and ``lxcXXXXXX`` in the host namespace. The router IP is allocated from the
+   ENI space, allowing for sending of ICMP errors from the router IP for Path
+   MTU purposes.
+
+2. After passing through the veth pair and before reaching the Linux routing
+   layer, all traffic is subject to Cilium's BPF program to enforce network
+   policies, implement load-balancing and provide networking features.
+
+3. An IP routing rule ensures that traffic from individual endpoints are using
+   a routing table specific to the ENI from which the endpoint IP was
+   allocated:
+
+   .. code-block:: bash
+
+       30:	from 192.168.105.44 to 192.168.0.0/16 lookup 92
+
+4. The ENI specific routing table contains a default route which redirects
+   to the router of the VPC via the ENI interface:
+
+   .. code-block:: bash
+
+       default via 192.168.0.1 dev eth2
+       192.168.0.1 dev eth2
+
+
+Configuration
+-------------
+
+The AWS ENI datapath is enabled by setting the following option:
+
+.. code-block: yaml
+
+        ipam: eni
+        blacklist-conflicting-routes: "false"
+        enable-endpoint-routes: "true"
+        auto-create-cilium-node-resource: "true"
+        egress-masquerade-interfaces: eth+
```

Comment:
Two thoughts here:

(1) Why can't we provide a global "ENI mode" setting that selects these micro settings automatically?

(2) As far as I know, `egress-masquerade-interfaces` is currently targeted to only work with the iptables-based masquerade and for kube-proxy-free will be superseded by `--devices: ...`. Not sure how we want to handle this potential mismatch. /cc @brb